src/bp/lang-server/api.ts

```diff
@@ -112,6 +112,19 @@ function createExpressApp(options: APIOptions): Application {
   return app
 }
 
+function assertLanguage(options: APIOptions, language: any) {
+  const availableLanguages = options.service.listFastTextModels().map(x => x.name)
+  if (!language) {
+    throw new BadRequestError(`Param 'lang' is mandatory`)
+  }
+  if (!_.isString(language)) {
+    throw new BadRequestError(`Param 'lang': ${language} must be a string`)
+  }
+  if (!availableLanguages.includes(language)) {
```

Comment:
When doing validations, try to avoid unnecessary processing. For example, if language isn't set or isn't a string, it will still list fast text models. I'd move the availableLanguage map just before it is being checked